### ISSUE TITLE
Ajouter des tests unitaires

### ISSUE THREAD

**Issue by strayMat** [SNDStoolers/sndsTools#2]
Nous devons effectuer des tests unitaires sur les fonctions existantes. Au moins un test de bout en bout pour chaque fonction. Il n'est pas nécessaire pour l'instant de tester chaque filtre / choix effectué par les fonctions d'extraction. 

Pour cela, plusieurs actions doivent être entreprises : 

- [x] créer des données de test, ce qui peut être facilité par l'instruction d'un LLM (chatgpt ou github copilot). Idéalement, les données sont spécifiques à chaque test unitaire. Cela permet de mieux dissocier les tests unitaires les uns des autres. 
- [x] Trouver un moyen de faire des tests unitaires sans avoir à instancier une base de données oracle. Justification : Nous devrions pouvoir développer et tester le package sans être sur le portail du CNAM. Mais, je n'ai pas trouvé de moyen facile d'installer ROracle sur mon bureau. Même s'il y a une installation simple, nous devrions instancier une base de données Oracle juste pour les tests, ce qui est pénible. Peut-être pourrions-nous forcer la fonction `initialize_connection` en mode local (sans être sur le portail CNAM) à retourner une connexion [sqlite](https://rsqlite.r-dbi.org/) où nous pourrions stocker les données de test dans un seul fichier `test.sqlite` ?  L'inconvénient serait d'ajouter une dépendance développeur (sqlite).

**Comment by strayMat** [comment]
Self-documentation sur les [bonnes pratique de tests en R](https://r-pkgs.org/testing-basics.html).

**Comment by strayMat** [comment]
J'ai un premier test fonctionnel en local. Cependant je dois adapter la syntaxe spécifique à oracle (TO_DATE) qui n'est pas comprise par sqlite pendant le testing.

**Comment by ThomasSoeiro** [comment]
La prise en charge des dates dans SQLite [est limitée](https://www.sqlite.org/datatype3.html#date_and_time_datatype). Ça pourra être embêtant selon les tests. Peut être à remplacer par [duckdb](https://github.com/duckdb/duckdb-r) ? C'est un système embarquée également et il y a un paquet R donc le changement est trivial.

**Comment by strayMat** [comment]
Merci pour le premier point, je tombais déjà sur des pbs avec sqlite... 

J'ai encore un problème avec la conversation TO_DATE d'oracle qui ne semble pas être supporté par duckdb non plus. 
Je pense qu'en cas de test je vais réécrire la function TO_DATE pour [strptime](https://duckdb.org/docs/sql/functions/timestamp.html#strptimetext-format) de duckdb

**Comment by ThomasSoeiro** [comment]
Si possible, tu peux utiliser les [`date literals`](https://docs.oracle.com/en/database/oracle/oracle-database/19/sqlrf/Literals.html#GUID-8F4B3F82-8821-4071-84D6-FBBA21C05AC1). C'est défini dans le standard ANSI et c'est pris en charge par Oracle et duckdb.

**Comment by ThomasSoeiro** [comment]
@strayMat Je reopen pour discuter de notre stratégie de test.

Je pense qu'il nous faut des tests mais que ceux qu'on a actuellement sont très peu utiles car sur des données fictives.
Il nous faudrait des tests à faire tourner sur le portail et vérifier que les résultats retournés soient cohérents.
Le portail est mouvant, il y a parfois des noms de variables qui changent par exemple.

Mais du coup ça ne peut pas être automatisé dans la CI...

**Comment by strayMat** [comment]
Je suis d'accord que des tests sur le portail seraient beaucoup plus utiles : je développe sur cette partie dans #17 

En revanche, je pense que des tests unitaires sur données fictives sont déjà une très bonne chose car ça détecte des problèmes grossiers qui feraient que la fonction planterait sur le portail, ça vérifie que les filtres que l'on fait marchent à peu près et ça facilite grandement le développement (en tout cas pour moi, qui écrit pas mal de code). 

Je suis en faveur de garder un test unitaire pour chaque fonction qui aide lors du développement et un test "data" sur le portail qui rassure sur la capacité à sortir ce que l'on veut comme données.

**Comment by ThomasSoeiro** [comment]
Je referme ce ticket pour continuer la discussion là bas, ce sera plus simple à suivre 😊
